plugins/modules/ec2_spot_instance.py

```diff
@@ -0,0 +1,359 @@
+    description:
+      - The type of spot request.
+      - After being interrupted a C(persistent) spot instance will be started once there is capacity to fill the request again.
+    default: "one-time"
+    choices: [ "one-time", "persistent" ]
+    type: str
+  valid_from:
+    description:
+      -  The start date of the request. If this is a one-time request, the request becomes active at this date and time and remains active until all instances launch, the request expires, or the request is canceled.
+      -  If the request is persistent, the request becomes active at this date and time and remains active until it expires or is canceled.      
+  valid_until:
+    description:
+      - The end date of the request
+  tags:
+    description:
+      - tag:value pairs to add to the volume after creation.
+    default: []
+    type: list
+  spot_instance_request_ids:
+    description:
+        - List of strings with IDs of spot requests to be cancelled
+    default: []
+    type: list
+'''
+
+EXAMPLES = '''
+# Simple Spot Request Creation
+- hosts: localhost
+  tasks:
+    - name: Test EC2 spot instance module
+      amazon.aws.ec2_spot_instance:
+        launch_specification:
+          image_id: "ami-123456789"
+          key_name: "my-keypair"
+          instance_type: "t2.medium"
+
+# Simple Spot Request Termination
+- hosts: localhost
+  tasks:
+    - name: Test EC2 spot instance cancel module
+      amazon.aws.ec2_spot_instance:
+        spot_instance_request_ids: ['sir-d8468pbj', 'sir-qph6aytk']
+        state: 'absent'
+'''
+
+RETURN = '''
+spot_request_info:
+    description: The spot instance request details after creation
+    returned: when success
+    type: dict
+    sample: {
+            "CreateTime": "2021-07-21T18:33:47+00:00",
+            "InstanceInterruptionBehavior": "terminate",
+            "LaunchSpecification": {
+                "ImageId": "ami-0d5eff06f840b45e9",
+                "InstanceType": "t2.medium",
+                "KeyName": "zuul",
+                "Monitoring": {
+                    "Enabled": false
+                },
+                "Placement": {
+                    "AvailabilityZone": "us-east-1e"
+                },
+                "SecurityGroups": [
+                    {
+                        "GroupId": "sg-0fa9a734e7111af56",
+                        "GroupName": "default"
+                    }
+                ],
+                "SubnetId": "subnet-0ff1d9ce7798affb1"
+            },
+            "ProductDescription": "Linux/UNIX",
+            "SpotInstanceRequestId": "sir-d8468pbj",
+            "SpotPrice": "0.046400",
+            "State": "open",
+            "Status": {
+                "Code": "pending-evaluation",
+                "Message": "Your Spot request has been submitted for review, and is pending evaluation.",
+                "UpdateTime": "2021-07-21T18:33:47+00:00"
+            },
+            "Type": "one-time"
+        }
+
+cancelled_spot_request_info:
```

Comment:
I wouldn't return this key.  For state=absent we can probably just return a `msg`.